File: Python3/Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        ans = {}
        for i in range(len(nums)):
            if nums[i] in ans:
                diff = abs(i - ans[nums[i]])
                if diff <= k:
                    return True
                else:
                    ans[nums[i]] = i
            else:
                ans[nums[i]] = i
        
        return False

File: Golang/Contains Duplicate II.go
func containsNearbyDuplicate(nums []int, k int) bool {
    ans := make(map[int]int)

    for i, v := range nums {
        if value, ok := ans[v]; ok {
            if absInt(i, value) <= k {
                return true
            } else {
                ans[v] = i
            }
        } else {
            ans[v] = i
        }
    }

    return false
}

func absInt(a, b int) int {
    if a - b < 0 {
        return b - a
    }
    
    return a - b
}